From 51fefb9546b272e18d8f13e5483fe4d4564e9203 Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Tue, 28 Apr 2020 23:09:38 +0800
Subject: [PATCH 012/122] fix typeo

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/others/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/release/src/router/others/Makefile b/release/src/router/others/Makefile
index d1acfe69b8..afe142f057 100755
--- a/release/src/router/others/Makefile
+++ b/release/src/router/others/Makefile
@@ -7,7 +7,7 @@ install:
 	install -d $(INSTALLDIR)/usr/sbin
 	install -d $(INSTALLDIR)/lib
 
-ifeq ($(TCONFIG_USB),y)
+ifeq ($(RTCONFIG_USB),y)
 # Optware perl symlink
 	install -d $(INSTALLDIR)/usr/bin
 	ln -sf /opt/bin/perl $(INSTALLDIR)/usr/bin/perl
-- 
2.25.1

