From bff448d158d86f594294043fb193c1b62f9e802a Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Thu, 30 Apr 2020 20:32:21 +0800
Subject: [PATCH 015/122] makefile: target remove bash for unused now

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/release/src/router/Makefile b/release/src/router/Makefile
index 55202042c0..00b1d15431 100644
--- a/release/src/router/Makefile
+++ b/release/src/router/Makefile
@@ -888,7 +888,7 @@ obj-y += others
 obj-y += libpasswd
 ifeq ($(RTK3),y)
 obj-y += k3screenctrl
-obj-y += bash
+#obj-y += bash
 endif
 obj-y += httpd json-c
 obj-$(RTCONFIG_VISUALIZATION) += sqlite3 json-c libxml2 visdcoll visdcon
@@ -2574,7 +2574,7 @@ roldconf: config/conf
 	$(MAKE) dnsmasq-clean iproute2-clean networkmap-clean
 ifeq ($(RTK3),y)
 	$(MAKE) k3screenctrl-clean
-	$(MAKE) bash-clean
+#	$(MAKE) bash-clean
 endif
 ifeq ($(RTCONFIG_BCMARM),y)
 	$(MAKE) compressed-clean
-- 
2.25.1

