From 2fcf253ec939c7e6797e27f508c2d7e78e4d757d Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Tue, 5 May 2020 12:33:42 +0800
Subject: [PATCH 018/122] makefile: include ROMCFE

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src-rt/target.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/release/src-rt/target.mak b/release/src-rt/target.mak
index 0389b8d5ad..73ecb2bd69 100644
--- a/release/src-rt/target.mak
+++ b/release/src-rt/target.mak
@@ -206,7 +206,7 @@ export RT-AC3100 += BUILD_NAME="RT-AC3100" SWITCH2="" BCM_MMC=n BCM_7114=y ETLAN
 		NEWSSID_REV2=y ASUSCTRL=y FRS_LIVE_UPDATE=n BW160M=y
 
 export RT-K3 := $(RT-AC88U_BASE)
-export RT-K3 += BUILD_NAME="RT-K3" SWITCH2="" BCM_MMC=n BCM_7114=y NOWLALL=n ETLAN_LED=n ROMCFE=n \
+export RT-K3 += BUILD_NAME="RT-K3" SWITCH2="" BCM_MMC=n BCM_7114=y NOWLALL=n ETLAN_LED=n ROMCFE=y \
 		NVSIZE="128" NOWL=y DHDAP=y DPSTA=y BCMFA=n GMAC3=y LACP=y WTFAST=y \
 		FORCE_SN="380" FORCE_EN="664" DISABLE_REPEATER_UI=n NEWSSID_REV2=y \
 		BRCM_NAND_JFFS2=n JFFS2LOG=n UBI=y UBIFS=y PSISTLOG=y
-- 
2.25.1

