From 7fe9195e9143a980def2235d16ae4487e22197a5 Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Tue, 5 May 2020 12:44:23 +0800
Subject: [PATCH 019/122] makefile: fix trx signature "RT-AC3100" for
 compatibility

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src-rt/Makefile | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/release/src-rt/Makefile b/release/src-rt/Makefile
index 6adda4137b..6297f10f01 100755
--- a/release/src-rt/Makefile
+++ b/release/src-rt/Makefile
@@ -243,6 +243,9 @@ ifeq ($(MODEL),RTK3)
 export RTK3=y
 export CFLAGS += -DRTK3
 MODEL:=RTAC3100
+export FILE_NAME:=RT-AC3100
+else
+export FILE_NAME:=$(BUILD_NAME)
 endif
 ifneq ($(MODEL),)
 export MODEL
@@ -990,9 +993,9 @@ else
 endif
 ifneq ($(CONFIG_BCMWL5),)
 ifeq ($(FAKEHDR),y)
-	ctools/trx_asus -i image/linux-lzma.trx -r $(BUILD_NAME),$(KERNEL_VER).$(FS_VER),$(FORCE_SN),$(FORCE_EN),$(SFINFO),image/$(IMGNAME).trx
+	ctools/trx_asus -i image/linux-lzma.trx -r $(FILE_NAME),$(KERNEL_VER).$(FS_VER),$(FORCE_SN),$(FORCE_EN),$(SFINFO),image/$(IMGNAME).trx
 else
-	ctools/trx_asus -i image/linux-lzma.trx -r $(BUILD_NAME),$(KERNEL_VER).$(FS_VER),$(SERIALNO),$(EXTENDNO),$(SFINFO),image/$(IMGNAME).trx
+	ctools/trx_asus -i image/linux-lzma.trx -r $(FILE_NAME),$(KERNEL_VER).$(FS_VER),$(SERIALNO),$(EXTENDNO),$(SFINFO),image/$(IMGNAME).trx
 endif
 else
 	ctools/trx_asus -i image/linux-lzma.trx -r $(BUILD_NAME),$(KERNEL_VER).$(FS_VER),image/$(IMGNAME).trx
-- 
2.25.1

