From 659dd3742095a011baa1d175c9a30b2b72726fca Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Thu, 15 Oct 2020 14:51:40 +0800
Subject: [PATCH 037/122] makefile: make generates CFE binary as well if
 ROMCFE=y.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src-rt/Makefile | 23 +++++++++++++++++++----
 1 file changed, 19 insertions(+), 4 deletions(-)

diff --git a/release/src-rt/Makefile b/release/src-rt/Makefile
index 6297f10f01..bb6b28a8a1 100755
--- a/release/src-rt/Makefile
+++ b/release/src-rt/Makefile
@@ -291,10 +291,15 @@ mips_rev =
 KERN_SIZE_OPT ?= y
 endif
 
+ifeq ($(RTK3)$(ROMCFE),yy)
+	export SUFFIX := _cfe
+else
+	export SUFFIX :=
+endif
 ifeq ($(FAKEID),y)
-export IMGNAME := $(BUILD_NAME)_$(FORCE_SN)_$(SWPJVER)$(FORCE_EN)$(BUILDREV)
+export IMGNAME := $(BUILD_NAME)_$(FORCE_SN)_$(SWPJVER)$(FORCE_EN)$(BUILDREV)$(SUFFIX)
 else
-export IMGNAME := $(BUILD_NAME)_$(SERIALNO)_$(SWPJVER)$(EXTENDNO)$(BUILDREV)
+export IMGNAME := $(BUILD_NAME)_$(SERIALNO)_$(SWPJVER)$(EXTENDNO)$(BUILDREV)$(SUFFIX)
 endif
 
 ifeq ($(DSL_REMOTE),y)
@@ -617,9 +622,14 @@ ifeq ($(RTCONFIG_REALTEK),y)
 	@echo "" > $(SRCBASE)/build.log
 endif
 	echo ""
-	echo "Building $(BUILD_NAME)_$(KERNEL_VER).$(FS_VER)_$(SERIALNO).trx"
+	echo "Building $(IMGNAME)"
 	echo ""
 
+ifeq ($(ROMCFE),y)
+	@$(MAKE) -C cfe clean
+	@$(MAKE) -C cfe
+endif
+
 ifeq ($(HND_ROUTER),y)
 	@rm -rf image
 	@ln -sf targets/${PROFILE} image
@@ -734,7 +744,12 @@ image:
 		exit 1; \
 	fi
 
-	@rm -f image/$(BUILD_NAME)_$(KERNEL_VER).$(FS_VER)_$(SERIALNO).trx
+	@echo "-------- clean old trx --------"
+ifeq ($(FAKEID),y)
+	rm -f image/$(BUILD_NAME)_$(FORCE_SN)_$(SWPJVER)$(FORCE_EN)$(BUILDREV)*.trx
+else
+	rm -f image/$(BUILD_NAME)_$(SERIALNO)_$(SWPJVER)$(EXTENDNO)$(BUILDREV)*.trx
+endif
 ifeq ($(RTCONFIG_REALTEK),y)
 ifeq ($(RTL8198D),y)
 	@rsync -aPq $(PLATFORMDIR)/target/ $(RSDKDIR)/romfs
-- 
2.25.1

