From 168b82573cd354171a034966926df2cbc1455cf8 Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Sun, 5 Jul 2020 08:05:04 +0800
Subject: [PATCH 054/122] rc: fix customize Tx power didn't work.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/rc/services.c | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/release/src/router/rc/services.c b/release/src/router/rc/services.c
index ceaec193df..fc44e8e49a 100644
--- a/release/src/router/rc/services.c
+++ b/release/src/router/rc/services.c
@@ -4772,18 +4772,20 @@ start_acsd()
 	stop_acsd();
 
 #ifdef RTK3
-	int tmpval;
+	char tmpval[3];
 	if (nvram_match("wl0_cpenable","1"))
 	{
-		tmpval = atoi(nvram_safe_get("wl0_custompower"));
-		printf("**** Set Tx power(2.4G): %d dbm\n", tmpval);
-		eval("wlk", "-i", "eth1", "txpwr1", "-o", "-d", MIN(tmpval, 31));
+		bzero(tmpval, sizeof(tmpval));
+		snprintf(tmpval, sizeof(tmpval), "%d", MIN(nvram_get_int("wl0_custompower"), 31));
+		printf("**** Set Tx power(2.4G): %s dbm\n", tmpval);
+		eval("wlk", "-i", "eth1", "txpwr1", "-o", "-d", tmpval);
 	}
 	if (nvram_match("wl1_cpenable","1"))
 	{
-		tmpval = atoi(nvram_safe_get("wl1_custompower"));
-		printf("**** Set Tx power(5G): %d dbm\n", tmpval);
-		eval("wlk", "-i", "eth2", "txpwr1", "-o", "-d", MIN(tmpval, 31));
+		bzero(tmpval, sizeof(tmpval));
+		snprintf(tmpval, sizeof(tmpval), "%d", MIN(nvram_get_int("wl1_custompower"), 31));
+		printf("**** Set Tx power(5G): %s dbm\n", tmpval);
+		eval("wlk", "-i", "eth2", "txpwr1", "-o", "-d", tmpval);
 	}
 #endif
 
-- 
2.25.1

