From 699755ccf886df415c27c87fc0d228a69f2bdb37 Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Sun, 5 Jul 2020 08:13:32 +0800
Subject: [PATCH 056/122] rc: optimize the code.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/rc/k3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/release/src/router/rc/k3.c b/release/src/router/rc/k3.c
index ddb104fae3..fbbab78789 100644
--- a/release/src/router/rc/k3.c
+++ b/release/src/router/rc/k3.c
@@ -142,7 +142,7 @@ void start_k3screen(void)
 
 	logmessage("K3INIT", "屏幕控制程序开始启动");
 	_dprintf("**** k3screen: start\n");
-	doSystem("mkdir -p /tmp/k3screenctrl");
+	mkdir_if_none("/tmp/k3screenctrl");
 	doSystem("ln -snf /lib/k3screenctrl/* /tmp/k3screenctrl");
 	_eval(k3screenbg_argv, NULL, 0, &pid1);
 	_eval(k3screenctrl_argv, NULL, 0, &pid2);
-- 
2.25.1

