From 186ca912a4eca1c76cbca72dcfc94c849c8800d8 Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Wed, 15 Jul 2020 01:39:59 +0800
Subject: [PATCH 060/122] target: update to 384.18

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src-rt/target.mak | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/release/src-rt/target.mak b/release/src-rt/target.mak
index 8dfc1f3eaa..dc05605a3a 100644
--- a/release/src-rt/target.mak
+++ b/release/src-rt/target.mak
@@ -206,9 +206,10 @@ export RT-AC3100 += BUILD_NAME="RT-AC3100" SWITCH2="" BCM_MMC=n BCM_7114=y ETLAN
 		NEWSSID_REV2=y ASUSCTRL=y FRS_LIVE_UPDATE=n BW160M=y
 
 export RT-K3 := $(RT-AC88U_BASE)
-export RT-K3 += BUILD_NAME="RT-K3" SWITCH2="" BCM_MMC=n BCM_7114=y NOWLALL=n ETLAN_LED=n ROMCFE=y \
-		NVSIZE="128" NOWL=y DHDAP=y DPSTA=y BCMFA=n GMAC3=y LACP=y WTFAST=y \
-		FORCE_SN="380" FORCE_EN="664" DISABLE_REPEATER_UI=n NEWSSID_REV2=y \
+export RT-K3 += BUILD_NAME="RT-K3" SWITCH2="" BCM_MMC=n BCM_7114=y ETLAN_LED=n ROMCFE=y \
+		NVSIZE="128" DHDAP=y DPSTA=y BCMFA=n GMAC3=y LACP=y WTFAST=y DWB=y \
+		DISABLE_REPEATER_UI=n NEWSSID_REV2=y ASUSCTRL=y FRS_LIVE_UPDATE=y \
+		BW160M=y AHS=n ASD=n INTERNETCTRL=y CAPTCHA=y WPS_DUALBAND=y\
 		BRCM_NAND_JFFS2=n JFFS2LOG=n UBI=y UBIFS=y PSISTLOG=y SOFTCENTER=y
 
 export RT-AC5300 := $(RT-AC88U_BASE)
-- 
2.25.1

