From 6490e6ba7e9e7b13089359fee828eedc651f8693 Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Wed, 15 Jul 2020 01:48:48 +0800
Subject: [PATCH 064/122] rc: remove some unused codes.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/rc/watchdog.c | 38 ++++++++++++--------------------
 1 file changed, 14 insertions(+), 24 deletions(-)

diff --git a/release/src/router/rc/watchdog.c b/release/src/router/rc/watchdog.c
index fb932d8af7..da1246797f 100644
--- a/release/src/router/rc/watchdog.c
+++ b/release/src/router/rc/watchdog.c
@@ -6557,31 +6557,21 @@ void dnsmasq_check()
 #ifdef RTK3
 void k3screen_check()
 {
-	if ((strcmp(nvram_get("k3screen"), "A")==0) || (strcmp(nvram_get("k3screen"), "a")==0))
+	if (!pids("k3screenbg"))
 	{
-		if (!pids("phi_speed"))
-			doSystem("phi_speed &");
-		if (!pids("wl_cr"))
-			doSystem("wl_cr &");
-		if (!pids("uhmi"))
-			doSystem("uhmi &");
-	} else {
-		if (!pids("k3screenbg"))
-		{
-			char *k3screend_argv[] = { "k3screenbg", NULL };
-			pid_t pid;
-			_eval(k3screend_argv, NULL, 0, &pid);
-			logmessage("watchdog", "restart k3screenbg");
-		}
-		if (!pids("k3screenctrl")){
-			char timeout[6];
-			int time = nvram_get_int("screen_timeout");
-			snprintf(timeout, sizeof(timeout), "-m%d", time);
-			char *k3screenctrl_argv[] = { "k3screenctrl", timeout, NULL };
-			pid_t pid;
-			_eval(k3screenctrl_argv, NULL, 0, &pid);
-			logmessage("watchdog", "restart k3screenctrl");
-		}
+		char *k3screend_argv[] = { "k3screenbg", NULL };
+		pid_t pid;
+		_eval(k3screend_argv, NULL, 0, &pid);
+		logmessage("watchdog", "restart k3screenbg");
+	}
+	if (!pids("k3screenctrl")){
+		char timeout[6];
+		int time = nvram_get_int("screen_timeout");
+		snprintf(timeout, sizeof(timeout), "-m%d", time);
+		char *k3screenctrl_argv[] = { "k3screenctrl", timeout, NULL };
+		pid_t pid;
+		_eval(k3screenctrl_argv, NULL, 0, &pid);
+		logmessage("watchdog", "restart k3screenctrl");
 	}
 }
 #endif
-- 
2.25.1

