From fd18feda6cea7484d7bd4a7f611c5e6539b6db4b Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Sun, 9 Aug 2020 18:25:00 +0800
Subject: [PATCH 075/122] www: add two missing variables for softcenter.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/www/state.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/release/src/router/www/state.js b/release/src/router/www/state.js
index f69928c986..241cd4f504 100644
--- a/release/src/router/www/state.js
+++ b/release/src/router/www/state.js
@@ -1262,6 +1262,8 @@ function submitenter(myfield,e)
 	else
 		return true;
 }
+var tabtitle = [""];
+var tablink = [""];
 
 function show_menu(){
 	var wan_pppoe_username = decodeURIComponent('<% nvram_char_to_ascii("", "wan0_pppoe_username"); %>');
-- 
2.25.1

