From cae9d74dde1588cbafa5d3a96358925b49d22c61 Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Sun, 24 Jan 2021 21:50:58 +0800
Subject: [PATCH 088/122] www: fix missing custom wl power.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 .../Advanced_WAdvanced_Content.asp            | 68 ++++++++++++++++++-
 1 file changed, 67 insertions(+), 1 deletion(-)

diff --git a/release/src/router/www/sysdep/FUNCTION/WL_SCHED_V2/Advanced_WAdvanced_Content.asp b/release/src/router/www/sysdep/FUNCTION/WL_SCHED_V2/Advanced_WAdvanced_Content.asp
index 72546281fd..155bffe0ac 100644
--- a/release/src/router/www/sysdep/FUNCTION/WL_SCHED_V2/Advanced_WAdvanced_Content.asp
+++ b/release/src/router/www/sysdep/FUNCTION/WL_SCHED_V2/Advanced_WAdvanced_Content.asp
@@ -89,6 +89,14 @@
 	border-bottom-right-radius: 1px;
 }
 #slider .ui-slider-handle { border-color: #93E7FF; }
+#slider1 .ui-slider-range {
+	background: #93E7FF;
+	border-top-left-radius: 3px;
+	border-top-right-radius: 1px;
+	border-bottom-left-radius: 3px;
+	border-bottom-right-radius: 1px;
+}
+#slider1 .ui-slider-handle { border-color: #93E7FF; }
 </style>
 <script>
 $(function () {
@@ -665,6 +673,7 @@ function initial(){
 	}
 
 	control_TimeField();
+	handle_chpower();
 
 	if(isSwMode("re")){
 		var _rows = document.getElementById("WAdvTable").rows;
@@ -781,11 +790,16 @@ function adjust_tx_power(){
 	var power_value_old = document.form.wl_TxPower.value;	//old nvram not exist now (value)
 	var power_value_new = document.form.wl_txpower.value;	//current nvram now (percentage)
 	var translated_value = 0;
-	
+	var power_value = parseInt(document.form.wl_custompower.value);
+
 	if(!power_support){
 		document.getElementById("wl_txPower_field").style.display = "none";
 	}
 	else{
+		document.getElementById('slider1').children[0].style.width = (power_value * 100 / 31) + "%";
+		document.getElementById('slider1').children[1].style.left = (power_value * 100 / 31) + "%";
+		document.getElementById("tx_power1_desc").innerHTML = power_value + "dbm " + Math.round(Math.pow(10,( power_value / 10 )))+ "mW";
+
 		if(power_value_old != ""){
 			translated_value = parseInt(power_value_old/80*100);
 			if(translated_value >=100){
@@ -838,6 +852,17 @@ function adjust_tx_power(){
 	}
 }
 
+function handle_chpower(){
+	if(document.form.wl_cpenable[0].checked){
+		document.getElementById("wl_custompower_field").style.display = "";
+		document.getElementById("wl_txPower_field").style.display = "none";
+	}
+	else{
+		document.getElementById("wl_custompower_field").style.display = "none";
+		document.getElementById("wl_txPower_field").style.display = "";
+	}
+}
+
 function changeRSSI(_switch){
 	if(_switch == 0){
 		document.getElementById("rssiDbm").style.display = "none";
@@ -1043,6 +1068,22 @@ function register_event(){
 			}
 		}); 
 	});
+
+	$(function() {
+		$( "#slider1" ).slider({
+			orientation: "horizontal",
+			range: "min",
+			min:0,
+			max: 31,
+			value:26,
+			slide:function(event, ui){
+				document.getElementById('tx_power1_desc').innerHTML = ui.value + "dbm " + Math.round(Math.pow(10,( ui.value / 10 )))+ "mW";
+			},
+			stop:function(event, ui){
+				document.form.wl_custompower.value = ui.value;
+			}
+		});
+	});
 }
 
 function set_power(power_value){	
@@ -1273,6 +1314,7 @@ function check_nodes_support_wireless_scheduler() {
 <input type="hidden" name="acs_dfs" value="<% nvram_get("acs_dfs"); %>">
 <input type="hidden" name="w_Setting" value="1">
 <input type="hidden" name="wl_txpower" value="<% nvram_get("wl_txpower"); %>">
+<input type="hidden" name="wl_custompower" value="<% nvram_get("wl_custompower"); %>">
 <input type="hidden" name="wl1_mumimo" value="<% nvram_get("wl1_mumimo"); %>" disabled>
 <table class="content" align="center" cellpadding="0" cellspacing="0">
 	<tr>
@@ -1639,6 +1681,13 @@ function check_nodes_support_wireless_scheduler() {
 							</select>
 						</td>
 					</tr>
+					<tr id="wl_chpower_enable">
+						<th><#WLANConfig11b_TxPower_custom#></th>
+						<td>
+							<input type="radio" value="1" name="wl_cpenable" class="input" onClick="handle_chpower();" <% nvram_match("wl_cpenable", "1", "checked"); %>><#checkbox_Yes#>
+							<input type="radio" value="0" name="wl_cpenable" class="input" onClick="handle_chpower();" <% nvram_match("wl_cpenable", "0", "checked"); %>><#checkbox_No#>
+						</td>
+					</tr>
 					<tr id="wl_txPower_field">
 						<th><a class="hintstyle" href="javascript:void(0);" onClick="openHint(0, 16);"><#WLANConfig11b_TxPower_itemname#></a></th>
 						<td>
@@ -1657,7 +1706,24 @@ function check_nodes_support_wireless_scheduler() {
 							</div>
 						</td>
 					</tr>
+					<tr id="wl_custompower_field">
+						<th><a class="hintstyle" href="javascript:void(0);" onClick="openHint(0, 16);"><#WLANConfig11b_TxPower_itemname#></a></th>
+						<td>
+							<div>
+								<table>
+									<tr>
+										<td style="border:0px;padding-left:0px;">
+											<div id="slider1" style="width:300px;"></div>
+										</td>
+										<td style="border:0px;width:60px;">
+											<div id="tx_power1_desc" style="width:150px;font-size:14px;"></div>
+										</td>
 
+									</tr>
+								</table>
+							</div>
+						</td>
+					</tr>
 					<!--QCA9984 platform only, e.g. BRT-AC828 -->
 					<tr>
 						<th><#WLANConfig11b_x_Hardware_Offloading#></th>
-- 
2.25.1

