From 616d46b2d825b56e61df8746256a802b81b96ada Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Mon, 8 Feb 2021 19:28:14 +0800
Subject: [PATCH 098/122] skipd: install to /usr/bin

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/release/src/router/Makefile b/release/src/router/Makefile
index 72dcb7f9e8..fc9aacea56 100644
--- a/release/src/router/Makefile
+++ b/release/src/router/Makefile
@@ -3671,10 +3671,10 @@ skipdbv2-clean:
 	@rm -f skipdbv2/Makefile
 
 skipdbv2-install: skipdbv2
-	install -D skipdbv2/bin/skipd $(INSTALLDIR)/skipdbv2/usr/sbin/skipd
-	install -D skipdbv2/bin/dbus $(INSTALLDIR)/skipdbv2/usr/sbin/dbus
-	$(STRIP) $(INSTALLDIR)/skipdbv2/usr/sbin/skipd
-	$(STRIP) $(INSTALLDIR)/skipdbv2/usr/sbin/dbus
+	install -D skipdbv2/bin/skipd $(INSTALLDIR)/skipdbv2/usr/bin/skipd
+	install -D skipdbv2/bin/dbus $(INSTALLDIR)/skipdbv2/usr/bin/dbus
+	$(STRIP) $(INSTALLDIR)/skipdbv2/usr/bin/skipd
+	$(STRIP) $(INSTALLDIR)/skipdbv2/usr/bin/dbus
 
 miniupnpd/config.h: shared/version.h
 ifeq ($(RTCONFIG_IGD2),y)
-- 
2.25.1

