From 6c3c0e42565c91058dca80b259f3778357e849ba Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Mon, 8 Feb 2021 19:34:34 +0800
Subject: [PATCH 100/122] init: move k3_init() to after all init.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/rc/init.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/release/src/router/rc/init.c b/release/src/router/rc/init.c
index 98a8bb529f..3b8ef589f0 100644
--- a/release/src/router/rc/init.c
+++ b/release/src/router/rc/init.c
@@ -11242,9 +11242,6 @@ int init_nvram(void)
 #endif
 	case MODEL_RTAC88U:
 	case MODEL_RTAC3100:
-#ifdef RTK3
-		k3_init();
-#endif
 		ldo_patch();
 
 		set_tcode_misc();
@@ -14882,6 +14879,9 @@ static void sysinit(void)
 #endif
 	restore_defaults(); // restore default if necessary
 	init_nvram2();
+#ifdef RTK3
+	k3_init();
+#endif
 
 #ifdef RPAX56
 	update_rf_para();
-- 
2.25.1

