From 88355aab3666793cdb4e2b1a9110981c158a1e33 Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Tue, 16 Feb 2021 12:52:17 +0800
Subject: [PATCH 107/122] usb2jffs: auto run softcenter script after usb mount.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/rc/usb.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/release/src/router/rc/usb.c b/release/src/router/rc/usb.c
index 68a085d670..0803b68610 100644
--- a/release/src/router/rc/usb.c
+++ b/release/src/router/rc/usb.c
@@ -2210,6 +2210,10 @@ _dprintf("usb_path: 4. don't set %s.\n", tmp);
 #ifdef RTCONFIG_SOFTCENTER
 			_dprintf("run scripts: jffsinit.sh\n");
 			eval("jffsinit.sh");
+			eval("/jffs/.asusrouter", "&");
+			eval("/koolshare/bin/ks-wan-start.sh", "start");
+			eval("/koolshare/bin/ks-services-start.sh", "start");
+			eval("/koolshare/bin/ks-nat-start.sh", "start_nat");
 #endif
 		}
 
-- 
2.25.1

