From a1ddfcf04d2d6da4f10e75e383c38fbef4da10ed Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Fri, 2 Apr 2021 20:52:32 +0800
Subject: [PATCH 115/122] rc: add back optimize_xbox option for RTAC3100.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/rc/init.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/release/src/router/rc/init.c b/release/src/router/rc/init.c
index d3674076fb..bfb1cfaa69 100644
--- a/release/src/router/rc/init.c
+++ b/release/src/router/rc/init.c
@@ -13127,7 +13127,7 @@ NO_USB_CAP:
 #ifdef RTCONFIG_BCMWL6
 	add_rc_support("wl6");
 #endif
-#if defined(RTN66U) || defined(RTAC66U) || defined(RTAC68U) || defined(DSL_AC68U) || defined(RTAC68U_V4) || defined(RTCONFIG_QCA)
+#if defined(RTN66U) || defined(RTAC66U) || defined(RTAC68U) || defined(DSL_AC68U) || defined(RTAC68U_V4) || defined(RTCONFIG_QCA) || defined(RTAC3100)
 #ifdef RTCONFIG_OPTIMIZE_XBOX
 	add_rc_support("optimize_xbox");
 #endif
-- 
2.25.1

