From 00ed37db9505f5028034eeda3b79b565fdb13b11 Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Fri, 2 Apr 2021 20:56:54 +0800
Subject: [PATCH 117/122] ubifs: make volume of ubifs easier.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/rc/ubifs.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/release/src/router/rc/ubifs.c b/release/src/router/rc/ubifs.c
index 629fb6122c..0c60aa45cc 100644
--- a/release/src/router/rc/ubifs.c
+++ b/release/src/router/rc/ubifs.c
@@ -200,7 +200,7 @@ void start_ubifs(void)
 
 			/* make ubi volume */
 			_dprintf("*** ubifs: create jffs2 volume\n");
-			eval("ubimkvol", UBI_DEV_PATH, "-s", vol_size_s, "-N", UBIFS_VOL_NAME);
+			eval("ubimkvol", UBI_DEV_PATH, "-N", UBIFS_VOL_NAME, "-m");
 		}
 	}
 #endif
@@ -281,12 +281,7 @@ void start_ubifs(void)
 	notice_set("ubifs", format ? "Formatted" : "Loaded");
 
 #if 1 /* enable legacy & asus autoexec */
-	if (((p = nvram_get("ubifs_exec")) != NULL) && (*p != 0)) {
-		chdir(UBIFS_MNT_DIR);
-		system(p);
-		chdir("/");
-	}
-	if (((p = nvram_get("jffs2_exec")) != NULL) && (*p != 0)) {
+	if (nvram_get_int("ubifs_exec") == 1 || nvram_get_int("jffs2_exec") == 1) {
 		chdir(UBIFS_MNT_DIR);
 		system(p);
 		chdir("/");
-- 
2.25.1

