From 3737bc19537ff5f1fa85d0ee04b2f5dbcd5ca58b Mon Sep 17 00:00:00 2001
From: Rokis <ghost1988102@gmail.com>
Date: Fri, 2 Apr 2021 21:00:25 +0800
Subject: [PATCH 119/122] k3: fix code error.

Signed-off-by: Rokis <ghost1988102@gmail.com>
---
 release/src/router/rc/k3.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/release/src/router/rc/k3.c b/release/src/router/rc/k3.c
index 954f5a9999..460e8fde60 100644
--- a/release/src/router/rc/k3.c
+++ b/release/src/router/rc/k3.c
@@ -401,13 +401,13 @@ void exec_tcplugin(void)
 			fprintf(fpmodel, nvram_get("productid"));
 			fclose(fpmodel);
 		}
-	}
-	if ((fpmac = fopen("/var/label_macaddr", "w")))
-	{
-		char *etmac = get_label_mac();
-		toLowerCase(etmac);
-		fprintf(fpmac, etmac);
-		fclose(fpmac);
+		if ((fpmac = fopen("/var/label_macaddr", "w")))
+		{
+			char *etmac = get_label_mac();
+			toLowerCase(etmac);
+			fprintf(fpmac, etmac);
+			fclose(fpmac);
+		}
 	}
 }
 #endif
-- 
2.25.1

